src/main/frontend/components/publishing.cljs

```diff
@@ -0,0 +1,189 @@
+          (fn [result]
+            (resolve result))
+          (fn [error]
+            (log/error :page/http-delete-failed error)
+            (reject error)))))))
+
+(defn update-state-and-notify
+  [page-name]
+  (page-handler/page-add-properties! page-name {:published false})
+  (notification/show! (util/format "Remove Page \"%s\" from Logseq server success" page-name) :success))
+
+(defn update-project
+  [project-name data]
+  (let [url (util/format "%sprojects/%s" config/api project-name)]
+    (js/Promise.
+      (fn [resolve reject]
+        (util/post url data
+          (fn [result]
+            (resolve result))
+          (fn [error]
+            (log/error :project/http-update-failed error)
+            (reject error)))))))
+
+(defn delete-project
+  [project-name]
+  (let [url (util/format "%sprojects/%s" config/api project-name)]
+    (js/Promise.
+      (fn [resolve reject]
+        (util/delete url
+          (fn [result]
+            (resolve result))
+          (fn [error]
+            (log/error :project/http-delete-failed error)
+            (reject error)))))))
+
+(defn project
+  [editor-state current-project pages]
+  (rum/with-context [[t] i18n/*tongue-context*]
+    (if (= :display @editor-state)
+      (when current-project
+        [:div.cp__publishing-pj
+         [:span.cp__publishing-pj-name current-project]
+         [:span.cp__publishing-edit
+          {:on-click
+           (fn [_]
+             (reset! editor-state :editor))}
+          (t :publishing/edit)]])
+      [:div.flex.cp__publishing_pj_edit
+       [:input#cp__publishing-project-input
+        {:placeholder current-project
+         :default-value current-project}]
+       [:div.cp__publishing-pj-bt
+        (ui/button
+          (t :publishing/save)
+          :on-click (fn [e]
+                      (util/stop e)
+                      (let [editor (.getElementById js/document "cp__publishing-project-input")
+                            v (.-value editor)
+                            data {:name v}]
+                        (-> (p/let [result (update-project current-project data)]
+                              (when (:result result)
+                                (state/update-current-project :name v)
+                                (notification/show! "Updated project name successfully." :success)
+                                (reset! editor-state :display)))
+                            (p/catch
+                              (fn [error]
+                                (notification/show! "Failed to updated project name." :failed))))))
+          :background "green")]
+
+       [:div.cp__publishing-pj-bt
+        (ui/button
+          (t :publishing/delete)
+          :on-click (fn [e]
+                      (util/stop e)
+                      (let [confirm-message
+                            (util/format
+                              "This operation will also delete all publishing under project \"%s\", continue?"
```

Comment:
Consider updating to "This operation will delete all the published pages under the project \"%s\", are you sure?"